### PR TITLE
Fix possible incorrect surface azimuth angle calculations when a surface has non convex vertice

### LINKED ISSUE #9906: Surface orientation starting vertex issues
Issue overview
--------------
In the attached [example file](https://github.com/TiejunWu/EnergyPlus-IDF-Files/blob/master/1ZoneUncontrolled_Modified_4.IDF), there is a single zone building. On one of the north facing wall, there is a window. The north facing wall was broken into two pieces.
image

The window and wall coordinates are:
image

image

The wall coordinate follows the counterclockwise direction. But it did not start from the top left corner,  and the first 3 vertices are not convex. When I run a simulation in EnergyPlus 22.2, I get the following error:

image
This error message shows that the program got the Azimuth angle of the wall to the opposite direction.

But the interesting thing is that on the south facing wall, I have the same configuration and sequence of vertices. That did not result in an error.

image

image

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows 10 64 Bit
 - Version of EnergyPlus: 22.2
 - Unmethours link or helpdesk ticket number N/A

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Defect file added (list location of defect file [here](https://github.com/TiejunWu/EnergyPlus-IDF-Files/blob/master/1ZoneUncontrolled_Modified_4.IDF))
 v23.1 file added to EnergyPlusDevSupport\DefectFiles
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #9906

 - When there are non-convex vertices in a surface, there is a possibility that the surface azimuth angle will be incorrectly computed by 180 degree (facing the other way). The current PR deals with that problem.

 - The severe warning message reported in #9906 is one of the consequences of having an incorrectly computed azimuth angle when dealing with a base surface (wall) and a subsurface (window). However, the incorrect azimuth angle(s) could have far more impacts later in the simulation process. The current PR fixes that incorrect calculation from the root.

- The fix mainly involves revisions and clean-ups on `DetermineAzimuthAndTilt()` function in the legacy code in Vector.cc

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Review by mjwitte — COMMENTED** [review]
@jcyuan2020 I can't say that I follow the changes 100%, but this fixes the error and does not introduce any CI diffs. 

Some nice cleanup of `DetermineAzimuthAndTilt`. I would go ahead and do more cleanup - delete any commented code that doesn't provide useful history, and delete the 2 unused parameters.

The new unit test needs a fix after merging in develop (`DataGlobalConstants --> Constant`).
Confirmed that the new unit test fails with develop, passes here.

**Review by mjwitte — APPROVED** [review]
@jcyuan2020 Additional cleanup looks good. When you are ready, mark for review and tag @Myoldmopar

**Comment by nrel-bot-2** [comment]
@jcyuan2020 it has been 7 days since this pull request was last updated.

File: src/EnergyPlus/SurfaceGeometry.cc [hunk 1]
```diff
@@ -1368,13 +1368,11 @@ namespace SurfaceGeometry {
                                                 state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).Sides,
                                                 state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).NewellSurfaceNormalVector);
                 DetermineAzimuthAndTilt(state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).Vertex,
-                                        state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).Sides,
```
Comment (Myoldmopar) [inline src/EnergyPlus/SurfaceGeometry.cc:1371]: Eliminating arguments, :+1: :+1:

File: src/EnergyPlus/Vectors.cc [hunk 2]
```diff
@@ -292,14 +292,12 @@ void VecRound(Vector &vec, Real64 const roundto)
     vec.z = nint64(vec.z * roundto) / roundto;
 }
 
-void DetermineAzimuthAndTilt(Array1D<Vector> const &Surf,       // Surface Definition
-                             [[maybe_unused]] int const NSides, // Number of sides to surface
```
Comment (Myoldmopar) [inline src/EnergyPlus/Vectors.cc:296]: Great to get rid of these unused args.

**Review by Myoldmopar — APPROVED** [review]
Everything looks OK here.  Thanks for the bit of extra cleanup.

**Comment by Myoldmopar** [comment]
OK local testing is proving fine, merging this in now.  Thanks @jcyuan2020

**Pull request merged**